src/emoji/data.js

```diff
@@ -28,12 +28,34 @@ export const getFilteredEmojis = (
   query: string,
   activeImageEmojiByName: $ReadOnly<{ [string]: ImageEmojiType }>,
 ): $ReadOnlyArray<{| emoji_type: EmojiType, name: string, code: string |}> => {
+  const matchingUnicodeEmoji = Object.entries(unicodeCodeByName).map(x => {
+    if (typeof x[1] !== 'string') {
+      // This is impossible, but flow isn't smart enough to figure that out.
+      throw new Error('Emoji name is not a string');
+    }
+    // We may want to have fancier emoji-matching logic in the future: this
+    // code will not work with skin-tone modifiers, for instance, and
+    // there may be other strange unicode things that warrant special
+    // handling. It would be simple to add code to strip off the
+    // skin-tone modifiers in the query, but it seems worse to suggest to
+    // the user the default skin color emoji when they've typed in a
+    // modified one than to display nothing. We should circle back to
+    // this later, and figure out what our overall strategy for skin-tone
+    // modifiers across the app should be.
```

Comment:
Yeah. We have an issue for supporting skin-tone modifiers, which this should link to: zulip/zulip#992.

I agree it's probably unhelpful to provide a partial match, like the version with the default/nonhuman skin tone when the user enters an emoji with a specific skin tone.

I think this *code* will actually work perfectly fine with skin-tone modifiers, once Zulip supports them in emoji reactions at all. When that happens, the "code" for the emoji will be something that `parseUnicodeEmojiCode` can turn into the actual sequence of codepoints: e.g. '1f44d-1f3fb' would be the code for 👍🏻. For the format of these codes, see the comment on `Reaction.emoji_code` here:
https://github.com/zulip/zulip/blob/15c4a72ac5297b1213ba93e986147c4c27a75af2/zerver/models.py#L2206-L2219
(That's linked to by the jsdoc on our `unicodeCodeByName`. There's probably other places we could usefully add links.)

Maybe what that boils down to is:
* This emoji-literal logic will only find exact matches. That's a product/UI choice we're making that deserves a comment calling it out explicitly.
* This means if the user enters an emoji that's in Unicode but not in Zulip, we won't show a match. One particular case of that is emoji with explicit skin tone (zulip/zulip#992); it would be simple to add logic to return the default-skin-tone version in that case, but it seems worse to do that than to display nothing.

Reply:
Yeah, I think the thought around this is mostly if #992 ends up resulting in having a setting for "default skin tone modifier", we might want to always include that in the list in addition to the literal match, even if they search for a different one? Or maybe not, I'm not super sure. But it seems like something to think about when we put in the work to decide how things should look for #992. In any case, I reworded the comment a bit to mention that.